packages/payment-detection/src/erc20/escrow-info-retriever.ts

```diff
@@ -0,0 +1,133 @@
+import { PaymentTypes } from '@requestnetwork/types';
+import { ethers } from 'ethers';
+import { getDefaultProvider } from '../provider';
+import { parseLogArgs } from '../utils';
+
+// The ERC20 escrow smart contract ABI fragment containing escrow specific events.
+const erc20EscrowContractAbiFragment = [
+  'event RequestFrozen(bytes indexed paymentReference)',
+  'event InitiatedEmergencyClaim(bytes indexed paymentReference)',
+  'event RevertedEmergencyClaim(bytes indexed paymentReference)',
+];
+
+/** Escrow contract event arguments. */
+type EscrowArgs = {
+  paymentReference: string;
+};
+
+/**
+ * Retrieves a list of payment events from a escrow contract.
+ */
+export default class EscrowERC20InfoRetriever
+  implements
+    PaymentTypes.IPaymentNetworkBaseInfoRetriever<
+      PaymentTypes.IPaymentNetworkBaseEvent<PaymentTypes.ESCROW_EVENTS_NAMES>,
+      PaymentTypes.ESCROW_EVENTS_NAMES
+    > {
+  public contractEscrow: ethers.Contract;
+  public provider: ethers.providers.Provider;
+
+  /**
+   * @param paymentReference The reference to identify the payment.
+   * @param escrowContractAddress The address of the escrow contract.
+   * @param escrowCreationBlockNumber The block that created the escrow contract.
+   * @param network The Ethereum network to use.
+   */
+  constructor(
+    private paymentReference: string,
+    private escrowContractAddress: string,
+    private escrowCreationBlockNumber: number,
+    private network: string,
+  ) {
+    // Creates a local or default provider.
+    this.provider = getDefaultProvider(this.network);
+
+    // Setup the ERC20 escrow contract interface.
+    this.contractEscrow = new ethers.Contract(
+      this.escrowContractAddress,
+      erc20EscrowContractAbiFragment,
+      this.provider,
+    );
```

Comment:
Is there a reason for not using the [`erc20EscrowToPayArtifact`](https://github.com/RequestNetwork/requestNetwork/blob/master/packages/smart-contracts/src/lib/artifacts/ERC20EscrowToPay/index.ts)? Is it because you need to be able to pass down a custom contract address, right?